src/templates/header.js

```diff
@@ -11,20 +11,91 @@ const links = [
   {title: 'Contact', link: '/contact'},
 ]
 
-const Header = ({ siteTitle }) => {
-  return (
-  <header>
-    <div>
-      <Nav 
-        logo={Logo({title: siteTitle})}
-        links={links}
-        GatsbyLink={Link}
-      />
-    </div>
-  </header>
-)
+class Header extends Component {
+
+  componentDidMount() {
+    if(typeof window !== 'undefined') {
+      let userDetails = {}
+      let SITE_NAME = window.location.origin;
+      document.getElementById('contact-form') &&
+      document
+        .getElementById('contact-form')
+        .addEventListener('submit', (e => {
```

Comment:
It is better if you disable the submit button once the form is submitted and re-enable it once the response is received